src/test/java/api/loans/EndPatronActionSessionTests.java

```diff
@@ -169,4 +202,90 @@ public void checkOutSessionIsNotEndedWhenCheckInSessionEnds()
     assertThat(patronSessionRecordsClient.getAll(), hasSize(1));
     assertThat(patronNoticesClient.getAll(), empty());
   }
+
+  @Test
+  public void checkInSessionShouldBeCreatedForCheckInByBarcode()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    IndividualResource james = usersFixture.james();
+    UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+    InventoryItemResource nod = itemsFixture.basedUponNod();
+
+    IndividualResource loan = loansFixture.checkOutByBarcode(nod, james);
+    loansFixture.checkInByBarcode(
+      new CheckInByBarcodeRequestBuilder()
+        .forItem(nod)
+        .at(checkInServicePointId));
+
+    assertThat(patronSessionRecordsClient.getAll(), Matchers.hasSize(2));
+
+    List<JsonObject> checkInSessions = getCheckInSessions();
+    assertThat(checkInSessions, Matchers.hasSize(1));
+
+    JsonObject checkInSession = checkInSessions.get(0);
+    assertThat(checkInSession.getString("patronId"), is(james.getId().toString()));
+    assertThat(checkInSession.getString("loanId"), is(loan.getId().toString()));
+  }
+
+  @Test
+  public void checkInSessionShouldNotBeCreatedWithoutLoanForCheckInByBarcode()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+    InventoryItemResource nod = itemsFixture.basedUponNod();
+
+    loansFixture.checkInByBarcode(
+      new CheckInByBarcodeRequestBuilder()
+        .forItem(nod)
+        .at(checkInServicePointId));
+
+    assertThat(patronSessionRecordsClient.getAll(), Matchers.empty());
```

Comment:
Personally I find assertions more readable when the matcher function is statically imported, it reduces the noise when trying to understand the intent.

```suggestion
    assertThat(patronSessionRecordsClient.getAll(), empty());
```

Reply:
updated